api/app/models/spree/api/webhooks/stock_item_decorator.rb

```diff
@@ -0,0 +1,20 @@
+module Spree
+  module Api
+    module Webhooks
+      module StockItemDecorator
+        def self.prepended(base)
+          base.after_update :queue_webhooks_requests_for_sold_out_products, if: :saved_changes?
+          base.after_destroy :queue_webhooks_requests_for_sold_out_products
+        end
+
+        private
+
+        def queue_webhooks_requests_for_sold_out_products
+          product.queue_webhooks_requests!('product.out_of_stock') unless product.in_stock?
```

Comment:
we should check if before the updated product was in stock, so we don't send this event multiple times

Reply:
@damianlegawiec could you explain it more, please?
We are checking here if product is still in stock after updating & deleting stock items

Reply:
@KacperMekarski yes, we only want to send this event when the product becomes out of stock when it was in stock before

Reply:
@damianlegawiec added spec to check that event won't be called for first default stock item:
https://github.com/spree/spree/pull/11460/files#diff-4fa30f2c51efd0cd81b7cfc6bfd4f6dd61b4d6ec3657671e5caa1735831473d5R56

Reply:
@KacperMekarski that's not what I've meant. 

Imagine a scenario: product is out of stock, there's a stock item update (whatever reason) that still persist 0 as stock, this will trigger this event and it shouldn't because it didn't change the found on hand